account_billing/models/account_billing.py

```diff
@@ -0,0 +1,202 @@
+# Copyright 2019 Ecosoft Co., Ltd (http://ecosoft.co.th/)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+
+from odoo import models, fields, api, _
+from odoo.exceptions import ValidationError
+
+MAP_INVOICE_TYPE_PARTNER_TYPE = {
+    'out_invoice': 'customer',
+    'in_invoice': 'supplier',
+}
+
+
+class AccountBilling(models.Model):
+    _name = 'account.billing'
+    _description = 'Account Billing'
+    _inherit = ['mail.thread']
+    _order = 'date desc, id desc'
+
+    name = fields.Char(
+        readonly=True,
+        copy=False,
+        help='Number of account.billing',
+    )
+    partner_id = fields.Many2one(
+        'res.partner',
+        string='Partner',
+        required=True,
+        default=lambda self: self._get_partner_id(),
+        help='Partner Information',
+        track_visibility='always'
+    )
+    date = fields.Date(
+        string='Billing Date',
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        default=fields.Date.context_today,
+        help='Effective date for accounting entries',
+        track_visibility='always'
+    )
+    due_date = fields.Date(
+        string='Due Date',
```

Comment:
Refering this https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#38fields
You don't need string, it is auto generated to 'Due Date' anyway.